src/Common/MessageBroker/Integration/AmqpLib/ConnectionFactory/AmqpStreamConnectionFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\MessageBroker\Integration\AmqpLib\ConnectionFactory;
+
+use Gaming\Common\MessageBroker\Exception\MessageBrokerException;
+use PhpAmqpLib\Connection\AbstractConnection;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use Throwable;
+
+final class AmqpStreamConnectionFactory implements ConnectionFactory
+{
+    public function __construct(
+        private readonly string $dsn
+    ) {
+    }
+
+    public function create(): AbstractConnection
+    {
+        $urlComponents = parse_url($this->dsn);
+        if (!is_array($urlComponents)) {
+            throw new MessageBrokerException('Cannot parse ' . $this->dsn . ' as url.');
+        }
+
+        try {
+            return new AMQPStreamConnection(
```

Comment:
Setup heartbeat for that connection.

Reply:
Blocked by #132. The default implementation shipped with amqp-lib only allows a single sigalrm handler.